types/tailwindcss/tailwind-config.d.ts

```diff
@@ -0,0 +1,1186 @@
+export type Variant =
```

Comment:
This appears to be a types-only file without a corresponding module in the underlying library.  Am I reading that correctly?  Is the goal just to prevent consumers from accidentally importing the types?  If not, why not just include them in index.d.ts?  Possible issues: if a consumer does import this file, will the import still be present at runtime?  What happens if the underlying library adds a module with this name (unlikely, since it doesn't seem to follow their naming conventions)?

Reply:
Separation of concerns

tailwind-config.d.ts is really big, and I did not want to pollute index.d.ts

This library does not export anything of the sort, I had to manually infer this from the documentation.

Reply:
Sorry, I'm not sure I understand.  What concerns are you separating?  It seems like tailwind-config.d.ts contains the types that are necessary for the declarations in index.d.ts.  If you don't want consumers to see the helper types, just don't export them.

I don't follow your point about webpack.  There's nothing wrong with adding types not found in the underlying library - that's the point of DefinitelyTyped.  My concern is with adding a _package_ not found in the underlying library.  Does webpack do that?

Reply:
This type definition has zero dependencies, so surely this would not be a problem?

Ok, I could move `tailwind-config.d.ts` into `index.d.ts` if that is better? what do you think?

Reply:
I'm not sure what you mean by "zero dependencies".  If you mean that it doesn't depend on anything, I don't think that's relevant to the file organization.  If you mean that nothing depends on it, I think that's only because it hasn't been merged yet - it seems to be quite a popular package.

Yes, as @peterblazejewicz and I have explained, having modules that don't exist in the underlying library makes it more likely that a user will consume it incorrectly and have problems at runtime.